### PR TITLE
Bxc 4375 refactor image server proxy

### PR DESCRIPTION
**Pull request by sharonluong**
This PR adds new tests and refactors the image server proxy code to more clearly separate the controller and service. This also affected some downloadImage code to use new Util code.
https://unclibrary.atlassian.net/browse/BXC-4375

### CODE REVIEW

**Review by bbpennel — CHANGES_REQUESTED** [review]
Looks good, mostly just suggestions aside from the releaseConnection bit

File: web-services-app/src/main/java/edu/unc/lib/boxc/web/services/rest/ImageServerProxyController.java [hunk 1]
```diff
@@ -76,41 +76,37 @@ public void getRegion(@PathVariable("id") String id,
...
+                return imageServerProxyService.streamJP2(id, region, size, rotation, quality, format);
             } catch (IOException e) {
                 LOG.error("Error retrieving streaming JP2 content for {}", id, e);
-                response.setStatus(HttpStatus.INTERNAL_SERVER_ERROR.value());
+                return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
             }
         } else {
             LOG.debug("Access was forbidden to {} for user {}", id, GroupsThreadStore.getUsername());
-            response.setStatus(HttpStatus.FORBIDDEN.value());
+            return new ResponseEntity<>(HttpStatus.FORBIDDEN);
```
[the remark this answers is not in the document]
Comment (sharonluong) [reply web-services-app/src/main/java/edu/unc/lib/boxc/web/services/rest/ImageServerProxyController.java:86]: yeah i had thought about doing that and had decided against it at first becuase i was complicating my other refactors. i'll go back and try it now

File: web-services-app/src/main/java/edu/unc/lib/boxc/web/services/rest/ImageServerProxyController.java [hunk 2]
```diff
@@ -30,87 +28,62 @@
...
-                        response.getOutputStream(), response, 1);
-            } catch (IOException e) {
-                LOG.error("Error retrieving streaming JP2 content for {}", id, e);
-                response.setStatus(HttpStatus.INTERNAL_SERVER_ERROR.value());
-            }
-        } else {
-            LOG.debug("Access was forbidden to {} for user {}", id, GroupsThreadStore.getUsername());
-            response.setStatus(HttpStatus.FORBIDDEN.value());
+        AccessGroupSet principals = getAgentPrincipals().getPrincipals();
+        accessControlService.assertHasAccess("Insufficient permissions to metadata for " + id,
```
Comment (bbpennel) [inline web-services-app/src/main/java/edu/unc/lib/boxc/web/services/rest/ImageServerProxyController.java:54]: I guess this message isn't really accurate for a request to get a jpg region

Comment (sharonluong) [reply web-services-app/src/main/java/edu/unc/lib/boxc/web/services/rest/ImageServerProxyController.java]: hehe whoops! fixed this and the missing word for the actual get metadata endpoint

**Review by bbpennel — APPROVED** [review]

**Pull request merged**